pilot/pkg/keycertbundle/watcher_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keycertbundle
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestWatcher(t *testing.T) {
+	watcher := NewWatcher()
+
+	// 1. no key cert bundle
+	watch1 := watcher.AddWatcher()
+	select {
+	case bundle := <-watch1:
+		t.Errorf("watched unexpected keyCertBundle: %v", bundle)
+		return
+	default:
+	}
+
+	key := []byte("key")
+	cert := []byte("cert")
+	ca := []byte("caBundle")
+	// 2. set key cert bundle
+	watcher.SetAndNotify(key, cert, ca)
+	select {
+	case <-watch1:
```

Comment:
nit: would add additional checks to ensure bundle retrieved is what was passed in SetAndNotify